xds/internal/client/transport_helper.go

```diff
@@ -246,16 +246,16 @@ func (t *TransportHelper) send(ctx context.Context) {
 			t.sendCh.Load()
 
 			var (
-				target         []string
-				rType          ResourceType
-				version, nonce string
-				send           bool
+				target                 []string
+				rType                  ResourceType
+				version, nonce, errMsg string
+				send                   bool
 			)
 			switch update := u.(type) {
 			case *watchAction:
 				target, rType, version, nonce = t.processWatchInfo(update)
 			case *ackAction:
-				target, rType, version, nonce, send = t.processAckInfo(update, stream)
+				target, rType, version, nonce, errMsg, send = t.processAckInfo(update, stream)
```

Comment:
Why does `processAckInfo` have to return the `errMsg`. Why can't we use it directly here? It is available in `update`.

Reply:
Right. This is unnecessary, and `processAckInfo` cannot change it anyway. Done.